homeassistant/bootstrap.py

```diff
@@ -103,6 +104,12 @@ def _setup_component(hass, domain, config):
             try:
                 config = component.CONFIG_SCHEMA(config)
             except vol.MultipleInvalid as ex:
+                if 'extra keys not allowed' in ex.error_message:
```

Comment:
Let's move this if/else construct to the `config_validation` helper.

```python
def log_exception(logger, exception, message):
    message = "{}: ".format(message)
    if 'extra keys not allowed' in ex.error_message:
        message += "X is an invalid option …".format(…)
    else:
        message += ex.error_message
    logger.error(message)
```

And then just
```python
cv.log_exception(…)
return False
```